File: arraysAndHashing/duplicates.py
# brutforce algo has a time complexity of O(n^2) and a space complexity of O(1)
# sorting algo has a time complexity of O(nlogn) and a space complexity of O(1)
# hashset algo has a time complexity of O(n) and a space complexity of O(n)

class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        hashset = set()
        for item in nums:
            if item in hashset:
                return True
            else:
                hashset.add(item)
        return False

File: arraysAndHashing/cppRealizations/duplicates.cpp
#include <set>
#include <vector>
#include <iostream>

using namespace std;

bool containsDuplicate(vector<int>& nums) {
    set<int> hashset;
    for(const int& item : nums){
        if(hashset.find(item) != hashset.end()){
            return true;
        }else{
            hashset.insert(item);
        }
    }
    return false;
}

int main(){
    vector<int> nums = {1,2,3,1};
    cout << containsDuplicate(nums) << endl;
    return 0;
}